sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -894,11 +894,20 @@ object CombineFilters extends Rule[LogicalPlan] with PredicateHelper {
 /**
  * Removes no-op SortOrder from Sort
  */
-object EliminateSorts  extends Rule[LogicalPlan] {
+object EliminateSorts  extends Rule[LogicalPlan] with PredicateHelper {
   def apply(plan: LogicalPlan): LogicalPlan = plan transform {
     case s @ Sort(orders, _, child) if orders.isEmpty || orders.exists(_.child.foldable) =>
       val newOrders = orders.filterNot(_.child.foldable)
       if (newOrders.isEmpty) child else s.copy(order = newOrders)
+    case s @ Sort(orders, _, p @ Project(projectList, grandChild)) =>
+      val aliasMap = AttributeMap(projectList.collect {
+        case a: Alias => (a.toAttribute, a.child)
+      })
+      val newOrders = orders
```

Comment:
i haven't actually look at the pr, but i think these few operations deserve a lot more comment than they have right now (zero).